code/modules/mapping/mapping_helpers.dm

```diff
@@ -101,3 +101,28 @@ GLOBAL_LIST_EMPTY(z_is_planet)
 	GLOB.z_is_planet["[T.z]"] = TRUE
 	return INITIALIZE_HINT_QDEL
 
+/obj/effect/mapping_helpers/template_spawner //spawning templates
+	name = "template spawner"
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "syndballoon"
+	layer = POINT_LAYER
+	var/list/template_name = list("Biological Storage Facility") //list of potential templates to pull from, uses the template's name instead of ID
+	var/map_templates = "space_ruins_templates" //the mapping template list from the mapping subsystem
+	var/datum/map_template/template_to_spawn
+
+/obj/effect/mapping_helpers/template_spawner/proc/get_template(id)
+	var/template_to_spawn = SSmapping.[map_templates][id]
```

Comment:
`SSmapping.[]` is probably not what you meant to write

Reply:
w h y d o e s i t c o m p i l e